apps/full-site-editing/full-site-editing-plugin/premium-content/premium-content.php

```diff
@@ -26,14 +26,47 @@
 require_once __DIR__ . '/subscription-service/include.php';
 
 /**
- * Registers all block assets so that they can be enqueued through the block editor
- * in the corresponding context.
+ * Register blocks and load block translations. If not done on init, the render
+ * callbacks of the dynamic blocks won't be executed in the front-end.
  *
  * @see    https://developer.wordpress.org/block-editor/tutorials/block-tutorial/applying-styles-with-stylesheets/
- * @throws RuntimeException If block assets files are not found.
+ * @see    https://github.com/Automattic/wp-calypso/pull/44825
  * @return void
  */
 function premium_content_block_init() {
+	register_blocks();
+	load_translations();
+}
+
+/**
+ * Action enqueue_block_editor_assets - Load static assets for the editor.
+ *
+ * @see    https://github.com/Automattic/wp-calypso/pull/44825
+ * @return void
+ */
+function premium_content_block_enqueue_block_editor_assets() {
+	load_static_editor();
+}
+
+/**
+ * Action enqueue_block_assets - Load static assets for the frontend.
+ *
+ * @see    https://github.com/Automattic/wp-calypso/pull/44825
+ * @return void
+ */
+function premium_content_block_enqueue_block_assets() {
+	load_static_frontend();
+}
```

Comment:
Is there a need to invoke `load_static_editor` and `load_static_frontend` instead of directly writing the code inside `premium_content_block_enqueue_block_editor_assets` and `premium_content_block_enqueue_block_assets` .

Reply:
Not really, inlined here: b740a677c6e